pinot-spi/src/main/java/org/apache/pinot/spi/executor/ExecutorServiceUtils.java

```diff
@@ -132,4 +133,14 @@ public static void close(ExecutorService executorService, long terminationMillis
       throw new RuntimeException(e);
     }
   }
+
+  /**
+   * Returns the hard limit of the number of threads that can be used by the multi-stage executor.
+   * @param config Pinot configuration
+   * @return hard limit of the number of threads that can be used by the multi-stage executor (no hard limit if <= 0)
+   */
+  public static int getMultiStageExecutorHardLimit(PinotConfiguration config) {
+    return config.getProperty(CommonConstants.Helix.CONFIG_OF_MULTI_STAGE_ENGINE_MAX_SERVER_QUERY_THREADS, 0)
+        * CommonConstants.Helix.MULTI_STAGE_ENGINE_MAX_SERVER_QUERY_THREADS_HARDLIMIT_FACTOR;
+  }
```

Comment:
It looks like this static method is bound to the hard limit executor. I don't think ExecutorServiceUtils is the correct place to have it. It would be more natural to be declared in the HardLimitExecutor class

Reply:
Yeah, I agree, moved it to the Executor itself.

a974159d74a63116ef234b0617ca60aaac5b042f